spmd/tensor/parallel/api.py

```diff
@@ -89,6 +94,92 @@ def _shard_self_attn_params(name: str, module: nn.Module) -> None:
                 module.register_module(n, tp_multi_head_attention)
 
 
+# Define partition functions needed to parallelize Linear modules
+def _rowwise_parallelize_fn(name, module, device_mesh):  # pyre-ignore[2, 3]
+    for name, param in module.named_parameters():
+        dist_spec = (  # type: ignore[list-item]
+            [Shard(1)] if name == "weight" else [Replicate()]
+        )
+        dist_param = torch.nn.Parameter(
+            distribute_tensor(param, device_mesh, dist_spec)
+        )
+        module.register_parameter(name, dist_param)
+
+
+def _colwise_parallelize_fn(name, module, device_mesh):  # pyre-ignore[2, 3]
+    for name, param in module.named_parameters():
+        dist_param = torch.nn.Parameter(
+            distribute_tensor(param, device_mesh, [Shard(0)])
+        )
+        module.register_parameter(name, dist_param)
+
+
+def _parallelize_linear(
+    module: nn.Module,
+    device_mesh: DeviceMesh,
+    parallel_style: ParallelStyle = ColwiseParallel(),
+    tp_mesh_dim: int = 0,
+) -> None:
+    """
+    This function requires that the input module be an object of :class:``nn.Linear``.
+    The module will be parallelized over the n-d :class:``DeviceMesh``
+    based on the :class:``ParallelStyle``.
+
+    Args:
+        module (nn.Module):
+            The :class:``nn.Module`` object to be parallelized.
+        device_mesh (DeviceMesh):
+            The n-d :class:``DeviceMesh`` over which the `module` will be parallelized.
+            If the mesh is more than 1-dimensional, we convert it to a 1-d :class:``DeviceMesh``
+            by `tp_mesh_dim`.
+        parallel_style (ParallelStyle):
+            :class:``ParallelStyle`` describes how weight and bias should be distributed
+            over :class:``DeviceMesh`` and how the input and output should be prepared for
+            Tensor Parallelism.
+            :class:``RowwiseStyle``: weight is sharded on dim 1 and bias is replicated.
+            :class:``ColwiseStyle``: weight and bias are both sharded on dim 0.
+            Default: :class:``ColwiseParallel``
+        tp_mesh_dim (int):
+            The dimension of :class:``DeviceMesh`` on which we perform
+            Tensor Parallelism.
+            Default: 0
+    Returns:
+        None
+    """
+
+    if not isinstance(module, nn.Linear):
+        raise RuntimeError(
+            f"Tensor parallel module expects a torch.nn.Linear module but received {type(module)}!"
```

Comment:
I got confusing by this saying "Tensor parallel module". Can we just "Expect ..."?